guide/slash-commands/response-methods.md

```diff
@@ -0,0 +1,230 @@
+
+Thankfully, Discord provides a way to hide resposne messages from everyone but the executor of the slash command. This type of message is called an `ephemeral` message and can be set by providing `ephemeral: true` in the `InteractionReplyOptions`, as follows:
+
+```js {5}
+client.on('interactionCreate', async interaction => {
+	if (!interaction.isChatInputCommand()) return;
+
+	if (interaction.commandName === 'ping') {
+		await interaction.reply({ content: 'Secret Pong!', ephemeral: true });
+	}
+});
+```
+
+Now when you run your command again, you should see something like this:
+
+<DiscordMessages>
+	<DiscordMessage profile="bot">
+		<template #interactions>
+			<DiscordInteraction
+				profile="user"
+				:command="true"
+				:ephemeral="true"
+			>ping</DiscordInteraction>
+		</template>
+		Pong!
+	</DiscordMessage>
+</DiscordMessages>
+
+Ephemeral responses are *only* available for interaction responses; another great reason to switch across to the new and improved features.
+
+## Editing responses
+
+After you've sent an initial response, you may want to edit that response for various reasons. This can be achieved with the `ChatInputCommandInteraction#editReply()` method:
+
+::: warning
+After the initial response, an interaction token is valid for 15 minutes, so this is the timeframe in which you can edit the response and send follow-up messages. You also **cannot** edit the ephemeral state of a message, so ensure that your first response sets this correctly.
+:::
+
+```js {1,8-9}
+const wait = require('node:timers/promises').setTimeout;
+
+client.on('interactionCreate', async interaction => {
+	if (!interaction.isChatInputCommand()) return;
+
+	if (interaction.commandName === 'ping') {
+		await interaction.reply('Pong!');
+		await wait(2000);
+		await interaction.editReply('Pong again!');
+	}
+});
+```
+
+In fact, editing your interaction response is necessary to [calculate the ping](popular-topics/faq#how-do-i-check-the-bot-s-ping) properly for this command.
+
+## Deferred responses
+
+As previously mentioned, Discord requires an acknowledgement from your bot within three seconds that the interaction was received. Otherwise, Discord considers the interaction to have failed and the token becomes invalid. But what if you have a command that performs a task which takes longer than three seconds before being able to reply?
+
+In this case, you can make use of the `ChatInputCommandInteraction#deferReply()` method, which triggers the `<application> is thinking...` message. This also acts as the initial response, to confirm to Discord that the interaction was received successfully. This allows you 15 minutes to complete your tasks before responding.
+<!--- here either display the is thinking message via vue-discord-message or place a screenshot -->
```

Comment:
if todo, mark as TODO: (requires package adaptation, etc.)
otherwise should be resolved before merge

Reply:
This is actually a copy paste from whatever was previously here. No idea if vue-discord-message supports it.

Reply:
doesn't look like it, start the comment with `TODO: ` then, please

```suggestion
<!--- TODO: Thinking... message, once available in components -->
```